### PR TITLE
pkg/sink(ticdc): iterate all Kafka configs to support KOP

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/8892

### What is changed and how it works?
Iterate all Kafka configs to support KOP.

#### Tests 

 - Manual test(see [this doc](https://pingcap.feishu.cn/docx/V6led4ovEoC40IxWM07cTllMnkg))
##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
 Iterate all config items to support KOP when getting the broker's configs
```

### CODE REVIEW

**Comment by hi-rustin** [comment]
In addition to these tests, I also tested the multi-topic feature.
1. Start Kop with docker-compose
2. Start TiDB cluster
3. Build and start TiCDC server
4. Create a changefeed with multi-topic config: `./cdc cli changefeed create --sink-uri="kafka://10.2.7.27:19092/testdsada?max-message-bytes=1048576&protocol=canal-json&kafka-version=2.9.0&replication-factor=1" --config=./changefeed.toml`
```toml
[sink]
enable-kafka-sink-v2 = true
dispatchers = [{matcher = ['*.*'], topic = "tidb_{schema}_{table}"}]
```
5. Write some data in upstream
```sql
mysql> use test;
Database changed
mysql> CREATE TABLE example1 (   id INT PRIMARY KEY,   b INT );
Query OK, 0 rows affected (0.15 sec)

mysql> INSERT INTO example1 (id,b) VALUES (2,2);
Query OK, 1 row affected (0.01 sec)

mysql> CREATE TABLE example2 (   id INT PRIMARY KEY,   b INT );
Query OK, 0 rows affected (0.09 sec)

mysql> INSERT INTO example2 (id,b) VALUES (2,2);
Query OK, 1 row affected (0.00 sec)
```
7. check the topic with kcat: 
```console
Metadata for all topics (from broker 1960160947: 10.2.7.27:19092/1960160947):
 1 brokers:
  broker 1960160947 at 10.2.7.27:19092 (controller)
 4 topics:
  topic "testdsada" with 3 partitions:
    partition 0, leader 1960160947, replicas: 1960160947, isrs: 1960160947
    partition 1, leader 1960160947, replicas: 1960160947, isrs: 1960160947
    partition 2, leader 1960160947, replicas: 1960160947, isrs: 1960160947
  topic "tidb_test_example2" with 3 partitions:
    partition 0, leader 1960160947, replicas: 1960160947, isrs: 1960160947
    partition 1, leader 1960160947, replicas: 1960160947, isrs: 1960160947
    partition 2, leader 1960160947, replicas: 1960160947, isrs: 1960160947
  topic "not_exist" with 3 partitions:
    partition 0, leader 1960160947, replicas: 1960160947, isrs: 1960160947
    partition 1, leader 1960160947, replicas: 1960160947, isrs: 1960160947
    partition 2, leader 1960160947, replicas: 1960160947, isrs: 1960160947
  topic "tidb_test_example1" with 3 partitions:
    partition 0, leader 1960160947, replicas: 1960160947, isrs: 1960160947
    partition 1, leader 1960160947, replicas: 1960160947, isrs: 1960160947
    partition 2, leader 1960160947, replicas: 1960160947, isrs: 1960160947
```
8. Check the data with kcat:
```json
kcat  -b 10.2.7.27:19092 -t tidb_test_example2 -o -10000
% Auto-selecting Consumer mode (use -P or -C to override)
{"id":0,"database":"test","table":"example2","pkNames":null,"isDdl":true,"type":"CREATE","es":1683358998100,"ts":1683358998598,"sql":"CREATE TABLE `example2` (`id` INT PRIMARY KEY,`b` INT)","sqlType":null,"mysqlType":null,"data":null,"old":null}
{"id":0,"database":"test","table":"example2","pkNames":["id"],"isDdl":false,"type":"INSERT","es":1683359005400,"ts":1683359005922,"sql":"","sqlType":{"id":4,"b":4},"mysqlType":{"id":"int","b":"int"},"old":null,"data":[{"id":"2","b":"2"}]}


kcat  -b 10.2.7.27:19092 -t tidb_test_example1 -o -10000
% Auto-selecting Consumer mode (use -P or -C to override)
{"id":0,"database":"test","table":"example1","pkNames":null,"isDdl":true,"type":"CREATE","es":1683358986400,"ts":1683358986592,"sql":"CREATE TABLE `example1` (`id` INT PRIMARY KEY,`b` INT)","sqlType":null,"mysqlType":null,"data":null,"old":null}
{"id":0,"database":"test","table":"example1","pkNames":["id"],"isDdl":false,"type":"INSERT","es":1683358992000,"ts":1683358992389,"sql":"","sqlType":{"id":4,"b":4},"mysqlType":{"id":"int","b":"int"},"old":null,"data":[{"id":"2","b":"2"}]}
```

**Review by nongfushanquan — APPROVED** [review]

File: pkg/sink/kafka/admin.go [hunk 1]
```diff
@@ -179,16 +179,21 @@ func (a *saramaAdminClient) GetBrokerConfig(
...
+	// 1. Kafka only returns requested configs.
+	// 2. Kop returns all configs.
+	for _, entry := range configEntries {
+		if entry.Name == configName {
+			return entry.Value, nil
+		}
 	}
 
-	return configEntries[0].Value, nil
+	log.Warn("Kafka config item not found",
```
Comment (asddongmen) [inline pkg/sink/kafka/admin.go:191]: Why do you just log error here?

**Review by asddongmen — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

File: pkg/sink/kafka/admin.go (hunk 1 above)
Comment (hi-rustin) [reply pkg/sink/kafka/admin.go:191]: Because sometimes caller will ignore this error.

**Comment by hi-rustin** [comment]
I tested that both "Standalone" and "Cluster" work.
1. Standalone: I tested that both "Standalone" and "Cluster" work.
2. Cluster: Standalone

**Pull request merged**